VmAgent.Core.UnitTests/BasicAssetExtractorTest.cs

```diff
@@ -29,62 +30,69 @@ public class BasicAssetExtractorTest
         private MultiLogger _multiLogger;
         private Mock<ISystemOperations> _mockSystemOperations;
 
-        private readonly string[] LinuxSupportedFileExtensions = 
+        private readonly string[] LinuxSupportedFileExtensions =
             { Constants.ZipExtension, Constants.TarExtension, Constants.TarGZipExtension };
 
-        private readonly string _TarAssetFileName = "testAsset.tar";
-        private readonly string _TarGzAssetFileName = "testAsset.tar.gz";
-        private readonly string _ZipAssetFileName = "testAsset.zip";
         private readonly string _targetFolder = @"/data/targetPath";
 
         [TestInitialize]
         public void BeforeEachTest()
         {
-            _multiLogger = new MultiLogger(NullLogger.Instance, new TelemetryClient(TelemetryConfiguration.CreateDefault()));
+            _multiLogger = new MultiLogger(NullLogger.Instance, new TelemetryClient(TelemetryConfiguration.CreateDefault()));      
             _mockSystemOperations = new Mock<ISystemOperations>();
             _basicAssetExtractor = new BasicAssetExtractor(_mockSystemOperations.Object, _multiLogger);
-            
-            _mockSystemOperations.Setup(x => x.IsOSPlatform(OSPlatform.Windows)).Returns(false);
         }
 
-
         [TestMethod, TestCategory("BVT")]
-        public void AssetExtractionFailedOnLinux()
+        [DataRow("testAsset.tar")]
+        public void AssetExtractionFailedOnLinux(string asset)
         {
+            SetUpMockOperationForTest(IsSetUpForWindows: false);
+
             _mockSystemOperations.Setup(x => x.RunProcessWithStdCapture(It.IsAny<Process>())).Returns((1, "stdOut", "stdErr"));
 
-            ExceptionAssert.Throws<Exception>(() => _basicAssetExtractor.ExtractAssets(_TarAssetFileName, _targetFolder));
+            ExceptionAssert.Throws<AssetExtractionFailedException>(() => _basicAssetExtractor.ExtractAssets(asset, _targetFolder));
         }
 
         [TestMethod, TestCategory("BVT")]
-        public void VerifyProcessStartInfoForTar()
+        [DataRow("")]
```

Comment:
I'm not sure if adding multiple DataRows is the best way. I checked how to use DynamicData to find a better way to provide the test dataset to each unit tests but seems it's unnecessary. Happy to hear any feedback!